### PR TITLE
Deprecate mobile/apps/auth

### PR DESCRIPTION
**Pull request by eessex**
- Removes `mobile/apps/auth`
- Moves logic for redirecting logged in users from `home` to `auth2`
- Moves routes for `log_in` and `sign_up` from `home` to `auth2`
- Removes references to `MOBILE_NEW_AUTH_MODAL`

Confirmed auth forms based on this list: https://docs.google.com/document/d/1Zyz7pe2aB9bPimbKYLPTPK_jCy0fzFKpjcsEhnk5uec/edit

### CODE REVIEW

**Comment by alloy** [comment]
:ceiling-cat: Is there still the possibility to rename `auth2` to `authentication` instead? The term ‘auth’ is overloaded and ambiguous, it’s helpful to use either ‘authentication’ or ‘authorization’.

**Comment by eessex** [comment]
Yeah that is totally possible, we are using that name already in Reaction. Planning to rename the desktop folder in a separate PR to keep changes as compact as possible.

File: src/desktop/apps/auth2/__tests__/routes.jest.ts [hunk 1]
```diff
@@ -238,4 +254,87 @@ describe("Routes", () => {
...
+      redirectLoggedInHome(req, res, next)
+      expect(res.redirect).toBeCalledWith("/")
+    })
+
+    it("redirects logged in users to home if they log in from /signup", () => {
+      req.get.mockReturnValueOnce("/signup")
+      req.url = "/signup"
+      req.user = {}
+      redirectLoggedInHome(req, res, next)
+      expect(res.redirect).toBeCalledWith("/")
```
Comment (l2succes) [inline src/desktop/apps/auth2/__tests__/routes.jest.ts]: Nice 👍 Thanks for the thorough tests @eessex

**Comment by l2succes** [comment]
Looks great  🎉feel free to merge after the directory renaming @eessex

**Comment by eessex** [comment]
Gonna go ahead and merge, as noted in the above comment I will rename in a separate PR and finish the rest of the auth cleanup there.

**Pull request merged**